src/components/application_manager/include/application_manager/core_service.h

```diff
@@ -0,0 +1,257 @@
+  void SetRemoteControl(bool enabled) FINAL;
+
+  /**
+   * @brief Is Remote Control allowed by Policy and User
+   */
+  bool IsRemoteControlAllowed() const FINAL;
+
+  /**
+   * Checks if application has remote control functions
+   * @param app application
+   * @return true if application has remote control functions
+   */
+  bool IsRemoteControlApplication(ApplicationSharedPtr app) const FINAL;
+
+  /**
+   * @brief Gets current state of the specified interface
+   * @param interface which state to get
+   * @return true if specified interface available otherwise false
+   */
+  bool IsInterfaceAvailable(
+      const HmiInterfaces::InterfaceID interface) const FINAL;
+
+  /**
+   * Removes fake parameters from request to HMI
+   * @param message message to handle
+   */
+  void RemoveHMIFakeParameters(application_manager::MessagePtr& message) FINAL;
+
+  /**
+   * @brief Get pointer to application by application id
+   * @param app_id application id
+   * return pointer to application
+   */
+  ApplicationSharedPtr GetApplication(ApplicationId app_id) FINAL;
+
+  /**
+   * @brief Send message to HMI
+   * @param message Message to HMI
+   */
+  void SendMessageToHMI(const MessagePtr& message) FINAL;
+
+  /**
+   * @brief Send message to mobile device
+   * @param message Message to mobile
+   */
+  void SendMessageToMobile(const MessagePtr& message) FINAL;
+
+  /**
+   * @brief Returns unique correlation ID for next HMI request
+   *
+   * @return Unique correlation ID
+   */
+  uint32_t GetNextCorrelationID() FINAL;
+
+  /**
+   * @brief Returns all applications
+   *
+   * @return List with shared pointers to applications
+   */
+  std::vector<ApplicationSharedPtr> GetApplications(AppExtensionUID uid) FINAL;
+
+  /**
+   * @brief Subscribes to notification from HMI
+   * @param hmi_notification string with notification name
+   */
+  void SubscribeToHMINotification(const std::string& hmi_notification) FINAL;
+
+  /**
+   * @brief Change hmi level of app and notify it
+   * @param app Application to be changed and notified
+   * @param level New HMI level of app
+   */
+  void ChangeNotifyHMILevel(ApplicationSharedPtr app,
+                            mobile_apis::HMILevel::eType level) FINAL;
+
+  const smart_objects::SmartObject* GetRCCapabilities() const FINAL;
+  /**
+   * @brief Notify HMI about app changing HMI Level
+   * only NONE, BACKGROUND and LIMITED levels are sent
+   * @param app Application to be changed and notified
+   * @param level New HMI level of app
+   */
+  void NotifyHMIAboutHMILevel(ApplicationSharedPtr app,
+                              mobile_apis::HMILevel::eType level) FINAL;
+
+  /**
+   * Gets all allowed module types
+   * @param app_id unique identifier of application
+   * @param list of allowed module types
+   * @return true if application has allowed modules
+   */
+  bool GetModuleTypes(const std::string& policy_app_id,
+                      std::vector<std::string>* modules) const FINAL;
+
```

Comment:
please add description for the new added functions

Reply:
Confirmed resolved 👍